clar2wasm/src/standard/standard.wat

```diff
@@ -1488,36 +1555,51 @@
         ;;   0..32 -> Initial hash vals (will be the result hash in the end)
         ;;   32..288 -> Space to store W (result of $block64)
         ;;   288..$length+288 -> shifted data
+
+        ;; Copying SHA-256's intial values to $stack-pointer
         (memory.copy (global.get $stack-pointer) (i32.const 0) (i32.const 32))
+        
+        ;; Leaving some space to store 64 words, then storing actual data
+        ;; 64 words of 4 bytes = 256 + 32 (bytes for intial values) = 288
         (memory.copy (i32.add (global.get $stack-pointer) (i32.const 288)) (local.get $offset) (local.get $length))
 
+        ;; Making sure that $res_len is divisible by 64 ($res_len%64==0)
         (local.set $res_len ;; total size of data with expansion
             (i32.add
                 (i32.or
                     ;; len + 8 bytes for the size
                     (i32.add (local.get $length) (i32.const 8))
+                    ;; 0x3f=63=111111 to make it divisible by 64
                     (i32.const 0x3f)
                 )
+                ;; 1 byte for '1' to add at the end of the data
                 (i32.const 1)
             )
         )
 
         ;; Add "1" at the end of the data
         (i32.store offset=288
             (i32.add (global.get $stack-pointer) (local.get $length))
+            ;; 0x80=128=10000000
             (i32.const 0x80)
         )
         
         ;; Fill the remaining part before the size with 0s
         (memory.fill
+            ;; 288+1('1' added after the data)=289
             (i32.add (i32.add (global.get $stack-pointer) (local.get $length)) (i32.const 289))
             (i32.const 0)
+            ;; Leave last 8 bytes for length as they are
             (i32.sub (i32.sub (local.get $res_len) (local.get $length)) (i32.const 8))
         )
 
-        ;; Add the size, as a 64bits big-endian integer
+        ;; Convert length to bits, (i.e. 13 bytes = 104 bits)
```

Comment:
I believe this is less clear than what was there before.

Reply:
Yes. That was unclear. Combined both